wavefront/resource_ingestion_policy.go

```diff
@@ -35,36 +51,126 @@ func resourceIngestionPolicy() *schema.Resource {
 				Type:     schema.TypeString,
 				Required: true,
 			},
+			ipAccountsKey: {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			ipGroupsKey: {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			ipSourcesKey: {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			ipNamespacesKey: {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			ipTagsKey: {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: ingestionPolicyTagSchema,
+				},
+			},
 		},
 	}
 }
 
-func resourceIngestionPolicyCreate(
-	d *schema.ResourceData, meta interface{}) error {
-	client := meta.(*wavefrontClient).client.IngestionPolicies()
+func ingestionPolicyTagSchema() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		ipKeyKey: {
+			Type:     schema.TypeString,
+			Required: true,
+		},
+		ipValueKey: {
+			Type:     schema.TypeString,
+			Required: true,
+		},
+	}
+}
+
+func flattenIngestionPolicyAccountIDs(accounts []wavefront.IngestionPolicyAccount) []string {
+	var tfMaps []string
```

Comment:
Is doing it in a case like this overkill / confusing?
```
var tags = make([]map[string]string, 0, len(raw.([]interface{})))
```

Generally seems like, if you are simply transforming an entire array then, regardless of type, doing something like this would be better?

Reply:
Yea, if you know up front what the size of your new slice will be, it makes since to specify that as the capacity of the new slice when you first create it. But in some cases, you may not know exactly how big your slice will be. Then it is ok to just start with the new slice being nil (empty).  In go, the zero value for a slice is nil. So declaring

var someSlice []int

someSlice will be nil (empty) initially.